### PR TITLE
feat hardis:org:community:update Add a command to activate or deactivate communities

### PR DESCRIPTION
**Pull request by tprouvot**
Hi @nvuillam,

We had the need on one of our project and I though that could be helpful for the community.
Let me know if the PR requires updates and adjustments.

Thanks !

### CODE REVIEW

**Comment by tprouvot** [comment]
I figured out that I was missing the doc, I'll add it

**Review by nvuillam — CHANGES_REQUESTED** [review]
We're almost goog, please check comments :)

**Comment by nvuillam** [comment]
> I figured out that I was missing the doc, I'll add it

the doc is automatically generated, don't worry , you input field description , that's large enough :)

File: src/commands/hardis/org/community/update.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+          uxLog(this, c.yellow(outputString));
+          return { outputString };
+        }
+      }
+
+      // Process Network update
+      const networkUpdates = networksQueryRes.records.map((network) => {
+        return { Id: network.Id, Status: status };
+      });
+      const bulkUpdateRes = await bulkUpdate('Network', 'update', networkUpdates, conn);
```
[the remark this answers is not in the document]
Comment (tprouvot) [reply src/commands/hardis/org/community/update.ts:92]: We can have more than one community to udpate that's a list

File: src/commands/hardis/org/community/update.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        bulkUpdateRes.failedResults.forEach((failedResult, index) => {
+          uxLog(this, c.red(`Error ${index + 1}: Network ID ${failedResult.sf__Id} failed to update. Reason: ${failedResult.sf__Error}`));
+        });
+      }
+
+      // Build results summary
+      uxLog(this, c.green(`${c.bold(updateSuccessNb)} network(s) were updated.`));
+
+      // Return an object to be displayed with --json
+      return {
```
[the remark this answers is not in the document]
Comment (tprouvot) [reply src/commands/hardis/org/community/update.ts:107]: Now you know which command I used to create this one 😄

**Comment by tprouvot** [comment]
Found a bug, let me fix it

File: src/commands/hardis/org/community/update.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+  public async run(): Promise<AnyJson> {
+    const { flags } = await this.parse(HardisOrgCommunityUpdate);
+    const networkNames = flags.name ? flags.name.split(',') : [];
+    const status = flags.status ? flags.status : '';
+    this.debugMode = flags.debug || false;
+
+    const conn = flags['target-org'].getConnection();
+
+    // Select networks that we want to update
```
Comment (tprouvot) [inline src/commands/hardis/org/community/update.ts:54]:
```suggestion
```

File: src/commands/hardis/org/community/update.ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    let networkIds: any[] = [];
+    if (networkNames.length === 0) {
+      uxLog(this, c.red(`Error: No network name(s) provided.`));
+    } else if (networkNames.length > 0) {
+      // Use includenetworks argument
```
Comment (tprouvot) [inline src/commands/hardis/org/community/update.ts:59]:
```suggestion
```

**Review by tprouvot — COMMENTED** [review]
Remove previous comments

File: src/commands/hardis/org/community/update.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (nvuillam) [reply src/commands/hardis/org/community/update.ts:94]: @tprouvot conn.update can take a list up to 200 (REST API behind) :)

**Review by nvuillam — APPROVED** [review]
Great, thank u very much for the PR :)

**Pull request merged**